### PR TITLE
fix check dims for ds with multiple vars

### LINKED ISSUE #760: raster dimension check fails for datasets of multiple variables with varying dimension size
### HydroMT version checks

- [X] I have checked that this issue has not already been reported.
- [X] I have checked that this bug exists on the latest version of HydroMT.

### Reproducible Example

raster._check_dimensions currently expect all variables to have the same dimensions.

```
from hydromt import raster
import xarray as xr

da2d = raster.full_from_transform(
        transform=[0.5, 0.0, 3.0, 0.0, -0.5, -9.0],
        shape=(4, 6),
        nodata=-1,
        name="test",
        crs=4326,
)
da3d = da2d.expand_dims("t")
da3d.name = "test_3d"
ds = xr.merge([da2d, da3d])
ds.raster._check_dimensions()
```

### Current behaviour

The example returns a ValueError while the dataset dimension  order is valid:

> ValueError: Invalid dimension order (('y', 'x')). You can use `obj.transpose(('y', 'x')) to reorder your dimensions.

### Desired behaviour

No ValueError for valid datasets

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by DirkEilander**
## Issue addressed
Fixes #760 

## Explanation
Relax assumption that all dataset variables must have the same dimensions.

## Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `main`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation if needed
- [x] Updated changelog.rst if needed
- [x] For predefined catalogs: update the catalog version in the file itself, the references in data/predefined_catalogs.yml, and the changelog in data/changelog.rst

## Additional Notes (optional)
Add any additional notes or information that may be helpful.

### CODE REVIEW

File: hydromt/raster.py [hunk 1]
```diff
@@ -654,24 +654,42 @@ def origin(self) -> tuple[float, float]:
...
-        da = self._obj[self.vars[0]] if isinstance(self._obj, xr.Dataset) else self._obj
-        extra_dims = [dim for dim in da.dims if dim not in dims]
+
+        def _check_dims(da: xr.DataArray, dims=dims) -> bool:
+            extra_dims = [dim for dim in da.dims if dim not in dims]
+            if len(extra_dims) == 1:
+                dims = tuple(extra_dims) + dims
+            elif len(extra_dims) > 1:
+                raise ValueError("Only 2D and 3D data arrays supported.")
+            return da.dims == dims, extra_dims
```
Comment (savente93) [inline hydromt/raster.py:664]: Please don't use closures if they are not necessary, they can cause very unexpected behavour. Could you also give this function  a more descriptive name that includes what it actually checks?

Comment (savente93) [reply hydromt/raster.py:664]: Also your type hint is incorrect, it returns a bool and I think a list of dimentions?

File: hydromt/raster.py [hunk 2]
```diff
@@ -654,24 +654,42 @@ def origin(self) -> tuple[float, float]:
...
+
+        extra_dims = []
+        if isinstance(self._obj, xr.DataArray):
+            check, dim0 = _check_dims(self._obj)
```
Comment (savente93) [inline hydromt/raster.py:668]: I think this name is very deceptive, since it's not just one dimention but a list of them.

**Review by hboisgon — APPROVED** [review]
LGTM!

File: hydromt/raster.py [hunk 3]
```diff
@@ -654,24 +654,42 @@ def origin(self) -> tuple[float, float]:
...
+            extra_dims.extend(dim0)
```
Comment (savente93) [inline hydromt/raster.py:669]: Do we want to extend here even if we raise a value error on the next line?

File: hydromt/raster.py [hunk 4]
```diff
@@ -654,24 +654,42 @@ def origin(self) -> tuple[float, float]:
...
+            if not check:
+                raise ValueError(
+                    f"Invalid dimension order ({self._obj.dims}). "
+                    f"You can use `obj.transpose({dims}) to reorder your dimensions."
+                )
+        else:
+            for var in self._obj.data_vars:
+                if not all([dim in self._obj[var].dims for dim in dims]):
+                    continue  # only check data variables with x and y dims
```
Comment (savente93) [inline hydromt/raster.py:678]: Does this check data vars that _have_ x and y dims or that _only have_ x and y dims?

**Review by savente93 — CHANGES_REQUESTED** [review]
functionality seems okay from what I can tell but I'd like to see a few code changes.

File: hydromt/raster.py (hunk 4 above)
Comment (DirkEilander) [reply hydromt/raster.py:678]: That check happens when getting the x and y dims in the raster.x_dim and raster.y_dim properties.

File: hydromt/raster.py (hunk 3 above)
Comment (DirkEilander) [reply hydromt/raster.py:669]: Only when no error is raised we actually use extra_dims, so I don't think this matters

File: hydromt/raster.py (hunk 2 above)
Comment (DirkEilander) [reply hydromt/raster.py:668]: will change the name

File: hydromt/raster.py (hunk 1 above)
Comment (DirkEilander) [reply hydromt/raster.py:664]: good suggestion

**Pull request merged**